.github/workflows/tests.yml

```diff
@@ -245,10 +247,19 @@ jobs:
         python-version: ['3.9']
         test-type: ['unit', 'integration']
         test-group: ['1', '2', '3']
+        # fewer test splits for quicker unit tests, keep 3 splits for integration
+        test-splits: [2, 3]
+        exclude:
+          - test-type: unit
+            test-group: '3'
+          - test-type: unit
+            test-splits: 3
+          - test-type: integration
+            test-splits: 2
```

Comment:
Oh to clarify, this basically binds the test type with the test split nummber, by excluding the pairs that we don't want, and then also corrects on the unit test type not needing to run a test group 3, right?